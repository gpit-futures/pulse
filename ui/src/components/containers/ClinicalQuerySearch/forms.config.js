export const valuesNames = {
  SELECT_AGE: 'selectAgeField',
  DATE_OF_BIRTH: 'dateOfBirth',
  AGE_RANGE: 'ageRange',
  MALE: 'sexMale',
  FEMALE: 'sexFemale',
  SEARCH_TYPE: 'type',
  QUERY_CONTAINS: 'queryContains',
  QUERY_TEXT: 'queryText',
  MIN_VALUE: 'minValue',
  MAX_VALUE: 'maxValue',
};

export const valuesLabels = {
  SELECT_AGE: 'Select Age Params',
  DATE_OF_BIRTH: 'Date of Birth',
  AGE_RANGE: 'Age Range',
  GENDER: 'Gender',
  SEARCH_TYPE: 'Search Type',
  QUERY_CONTAINS: 'Search Query',
  MALE: 'Male',
  FEMALE: 'Female',
};

export const optionsForAgeField = [{
  value: 'birthday',
  title: 'Date of Birth',
}, {
  value: 'range',
  title: 'Age Range',
}];

export const optionsForSearchType = [
  { title: 'Allergies', value: 'allergies' },
  { title: 'Problems / Diagnosis', value: 'diagnosis' },
  { title: 'Procedures', value: 'procedures' },
  { title: 'Medications', value: 'medications' },
];

export const optionsForSearchQuery = [
  { title: 'contains', value: true },
];

