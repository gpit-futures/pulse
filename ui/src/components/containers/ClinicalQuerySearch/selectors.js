import { getFormValues, getFormMeta, isValid } from 'redux-form'
import { createSelector } from 'reselect';

const formValuesSelector = state => getFormValues('clinicalQuerySearchForm')(state);
const formMetaSelector = state => getFormMeta('clinicalQuerySearchForm')(state);
const formIsValidSelector = state => isValid('clinicalQuerySearchForm')(state);

const formStateSelector = createSelector(
  formValuesSelector,
  formMetaSelector,
  formIsValidSelector,
  (formValues, formMeta, formIsValid) => ({ formValues, formMeta, formIsValid })
);

export default formStateSelector;
