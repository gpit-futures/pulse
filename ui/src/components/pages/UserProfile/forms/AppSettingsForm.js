import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateAppSettingsForm } from '../forms.validation';
import { valuesSettingsForm, valuesSettingsFormLabels } from '../forms.config';
import { setTheme } from '../../../../ducks/set-theme.duck';

@reduxForm({
  form: 'appSettingsFormSelector',
  validate: validateAppSettingsForm,
})
export default class AppSettingsForm extends PureComponent {
  static propTypes = {
    dispatch: PropTypes.func,
    initialize: PropTypes.func,
  };

  componentDidMount() {
    const { patientsInfo, initialize } = this.props;
    initialize(this.defaultValuesForm(patientsInfo));
  }

  defaultValuesForm(value) {
    const defaultFormValues = {
      [valuesSettingsForm.APP_TITLE]: value.title,
      [valuesSettingsForm.LOGO_PATH]: value.logoB64,
      [valuesSettingsForm.SELECT_THEME]: value.themeColor,
      [valuesSettingsForm.BROWSER_TITLE]: value.browserTitle,
    };

    return defaultFormValues;
  }

 setThemeHook = dispatch => (theme) => {
   dispatch(setTheme(theme));
   return theme;
 };

 render() {
   const { dispatch } = this.props;

   return (
     <div className="panel-body-inner">
       <form name="appSettingsForm" className="form">
         <div className="form-group-wrapper">
           <Row>
             <Col xs={12} md={6}>
               <Row>
                 <Col md={11}>
                   <Field
                     label={valuesSettingsFormLabels.APP_TITLE}
                     name={valuesSettingsForm.APP_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                   <Field
                     label={valuesSettingsFormLabels.LOGO_PATH}
                     name={valuesSettingsForm.LOGO_PATH}
                     component={FileInput}
                     id={valuesSettingsForm.LOGO_PATH}
                   />
                   <Field
                     label={valuesSettingsFormLabels.SELECT_THEME}
                     name={valuesSettingsForm.SELECT_THEME}
                     component={Select}
                     options={optionsForThemesField}
                     //here we're using normalize hook for dispatching action to set app theme
                     normalize={this.setThemeHook(dispatch)}
                   />
                   <Field
                     label={valuesSettingsFormLabels.BROWSER_TITLE}
                     name={valuesSettingsForm.BROWSER_TITLE}
                     type="text"
                     placeholder=""
                     component={ValidatedInput}
                   />
                 </Col>
               </Row>
             </Col>
           </Row>
         </div>
       </form>
     </div>)
 }
}
