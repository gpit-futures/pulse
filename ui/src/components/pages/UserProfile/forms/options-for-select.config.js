const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

const optionsForCountryField = [{
  value: 'usa',
  title: 'USA',
}, {
  value: 'country1',
  title: 'Country 1',
}, {
  value: 'country2',
  title: 'Country 2',
}, {
  value: 'country3',
  title: 'Country 3',
}, {
  value: 'country4',
  title: 'Country 4',
}, {
  value: 'country5',
  title: 'Country 5',
}];

export { optionsForThemesField, optionsForGenderField, optionsForCountryField }
