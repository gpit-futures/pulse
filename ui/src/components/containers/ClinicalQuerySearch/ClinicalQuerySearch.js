import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import classNames from 'classnames';
import qs from 'qs';
import { connect } from 'react-redux'

import ClinicalQuerySearchForm from './ClinicalQuerySearchForm/ClinicalQuerySearchForm';
import formStateSelector from './selectors';
import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';
import { clientUrls } from '../../../config/client-urls.constants';
import { valuesNames, valuesLabels } from './forms.config';

@connect(formStateSelector)
export default class ClinicalQuerySearch extends PureComponent {
  static propTypes = {
    onClose: PropTypes.func.isRequired,
    formValues: PropTypes.object,
    formIsValid: PropTypes.bool,
  };

  static contextTypes = {
    router: PropTypes.shape({
      history: PropTypes.object,
    }),
  };

  state = {
    isOpen: true,
  };

  hideForm = () => this.setState({ isOpen: false });

  toggleFormVisibility = () => this.setState(prevState => ({ isOpen: !prevState.isOpen }));

  formValuesToSearchString = (formValues) => {
    const isDateOfBirthValid = !_.isEmpty(formValues[valuesNames.DATE_OF_BIRTH]);
    const sendData = {};

    sendData[valuesNames.QUERY_CONTAINS] = formValues[valuesNames.QUERY_CONTAINS];
    sendData[valuesNames.QUERY_TEXT] = formValues[valuesNames.QUERY_TEXT];
    sendData[valuesNames.MALE] = formValues[valuesNames.MALE];
    sendData[valuesNames.FEMALE] = formValues[valuesNames.FEMALE];
    sendData[valuesNames.SEARCH_TYPE] = formValues[valuesNames.SEARCH_TYPE];

    if (isDateOfBirthValid) {
      sendData[valuesNames.DATE_OF_BIRTH] = formValues[valuesNames.DATE_OF_BIRTH];
    } else {
      sendData[valuesNames.MIN_VALUE] = formValues[valuesNames.AGE_RANGE][0];
      sendData[valuesNames.MAX_VALUE] = formValues[valuesNames.AGE_RANGE][1];
    }

    return sendData;
  };

  formValuesToTitle = (formValues) => {
    const isAgeRangeSelected = _.flow(_.get(valuesNames.SELECT_AGE), _.eq('range'))(formValues);

    const ageRangeTitle = _.flow(_.getOr('', valuesNames.AGE_RANGE), _.cond([
      [_.isEmpty, _.constant('')],
      [_.T, (ageRange => `${valuesLabels.AGE_RANGE}: ${ageRange[0]}-${ageRange[1]}`)],
    ]))(formValues);

    const dateOfBirth = _.flow(_.getOr('', valuesNames.DATE_OF_BIRTH), _.cond([
      [_.isEmpty, _.constant('')],
      [_.T, (date => `${valuesLabels.DATE_OF_BIRTH}: ${getDDMMMYYYY(date)}`)],
    ]))(formValues);

    const genderTitle = _.flow(values => ({
      isMale: _.getOr('', valuesNames.MALE, values),
      isFemale: _.getOr('', valuesNames.FEMALE, values),
    }), _.cond([
      [({ isMale, isFemale }) => isMale && isFemale, _.constant(`${valuesLabels.GENDER}: All`)],
      [({ isMale }) => isMale, _.constant(`${valuesLabels.GENDER}: Male`)],
      [({ isFemale }) => isFemale, _.constant(`${valuesLabels.GENDER}: Female`)],
      [_.T, _.constant('')],
    ]))(formValues);

    const typeTitle = _.flow(_.getOr('', valuesNames.SEARCH_TYPE), _.cond([
      [_.isEmpty, _.constant('')],
      [_.T, (type => `${valuesLabels.SEARCH_TYPE}: ${type}`)],
    ]))(formValues);

    const title = [typeTitle, isAgeRangeSelected ? ageRangeTitle : dateOfBirth, genderTitle].filter(_.negate(_.isEmpty)).join(', ');

    return title
      ? `: ${title}`
      : '';
  };

  handleSearch = (e) => {
    e.preventDefault();
    const { formValues, formIsValid } = this.props;
    const sendFormValue = {
      [valuesNames.QUERY_CONTAINS]: formValues[valuesNames.QUERY_CONTAINS],
      [valuesNames.SELECT_AGE]: formValues[valuesNames.SELECT_AGE],
      [valuesNames.FEMALE]: formValues[valuesNames.FEMALE],
      [valuesNames.MALE]: formValues[valuesNames.MALE],
      [valuesNames.SEARCH_TYPE]: formValues[valuesNames.SEARCH_TYPE],
      [valuesNames.QUERY_TEXT]: formValues[valuesNames.QUERY_TEXT],
    };

    if (formIsValid) {
      if (formValues.selectAgeField === 'range') {
        sendFormValue[valuesNames.AGE_RANGE] = formValues[valuesNames.AGE_RANGE]
      }
      if (formValues.selectAgeField === 'birthday') {
        sendFormValue[valuesNames.DATE_OF_BIRTH] = formValues[valuesNames.DATE_OF_BIRTH]
      }
      const queryParams = {
        searchString: JSON.stringify(this.formValuesToSearchString(sendFormValue)),
        queryType: 'clinicalQuery',
      };

      const searchReportUrl = `${clientUrls.SEARCH_REPORT}?${qs.stringify(queryParams)}`;
      this.hideForm();
      this.context.router.history.push(searchReportUrl);
    }
  };

  render() {
    const { isOpen } = this.state;
    const { onClose, formValues, formIsValid } = this.props;

    return (
      <div className={classNames('dropdown-menu dropdown-menu-search dropdown-menu-right', { 'without-shadow': isOpen })}>
        <div className="panel-group accordion">
          <div className={classNames('panel panel-secondary without-margin', { open: isOpen })}>
            <div className="panel-heading">
              <div className="control-group right">
                <button className="btn btn-success btn-inverse btn-square btn-toggle-rotate" onClick={this.toggleFormVisibility}>
                  <i className="btn-icon fa fa-chevron-up" />
                </button>
              </div>
              <h3 className="panel-title">
                <span className="ng-binding">Clinical Query</span>
                <span className="hidden-xs hidden-sm ng-binding">{this.formValuesToTitle(formValues)}</span>
              </h3>
            </div>
            <div className="panel-body" style={{ display: isOpen ? 'block' : 'none' }}>
              <div className="panel-body-inner">
                <ClinicalQuerySearchForm formValues={formValues} handleSearch={this.handleSearch} />
              </div>
              <div className="panel-control">
                <div className="wrap-control-group hide-indent-bottom">
                  <div className="control-group with-indent right">
                    <button className="btn btn-danger btn-icon-normal" onClick={onClose}><i className="btn-icon fa fa-times" /> <span className="btn-text">Close</span></button>
                    <button className="btn btn-success btn-icon-normal" disabled={!formIsValid} onClick={this.handleSearch}>
                      <i className="btn-icon fa fa-search" />
                      <span className="btn-text">Search</span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}
