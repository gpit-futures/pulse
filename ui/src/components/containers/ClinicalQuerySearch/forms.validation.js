import _ from 'lodash/fp';

import { valuesNames } from './forms.config';

export const validateForm = (values) => {
  const errors = {};
  const isAgeRangeSelected = _.flow(_.get(valuesNames.SELECT_AGE), _.eq('range'))(values);

  errors[valuesNames.SEARCH_TYPE] = !values[valuesNames.SEARCH_TYPE] ? 'You must enter a value.' : null;
  errors[valuesNames.QUERY_TEXT] = !values[valuesNames.QUERY_TEXT] ? 'You must enter a value.' : null;
  if (!isAgeRangeSelected) errors[valuesNames.DATE_OF_BIRTH] = !values[valuesNames.DATE_OF_BIRTH] ? 'You must enter a value.' : null;

  return errors
};
