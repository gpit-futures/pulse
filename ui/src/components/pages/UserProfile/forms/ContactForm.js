import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from '../forms.validation';
import { valuesContactForm, valuesContactFormLabels } from '../forms.config';
import { defaultContactFormValues } from './default-values.config';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  componentDidMount() {
    this.props.initialize(defaultContactFormValues);
  }

  render() {
    return (
      <div className="panel-body-inner">
        <form name="contactForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label={valuesContactFormLabels.ADDRESS}
                      name={valuesContactForm.ADDRESS}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.CITY}
                      name={valuesContactForm.CITY}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.STATE}
                      name={valuesContactForm.STATE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.POSTAL_CODE}
                      name={valuesContactForm.POSTAL_CODE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.SELECT_COUNTRY}
                      name={valuesContactForm.SELECT_COUNTRY}
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label={valuesContactFormLabels.PHONE}
                      name={valuesContactForm.PHONE}
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label={valuesContactFormLabels.EMAIL}
                      name={valuesContactForm.EMAIL}
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
