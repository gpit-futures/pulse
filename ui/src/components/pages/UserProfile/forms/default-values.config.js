import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from '../forms.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
  [valuesPersonalForm.DATE_OF_BIRTH]: 0,
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
