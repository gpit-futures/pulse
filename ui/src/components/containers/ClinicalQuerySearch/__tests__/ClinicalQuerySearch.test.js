import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import ClinicalQuerySearch from '../ClinicalQuerySearch';
import { valuesNames } from '../forms.config';


Enzyme.configure({ adapter: new Adapter() });

Date.now = jest.fn(() => new Date(Date.UTC(2017, 7, 9, 8)).valueOf())

const mockStore = configureStore();
const storeResource = {
  form: {
    clinicalQuerySearchForm: {
      values: {
        [valuesNames.SELECT_AGE]: 'range',
        [valuesNames.AGE_RANGE]: [
          0,
          100
        ],
        [valuesNames.MALE]: false,
        [valuesNames.FEMALE]: false,
        [valuesNames.QUERY_CONTAINS]: true,
        [valuesNames.SEARCH_TYPE]: 'allergies',
        [valuesNames.SEARCH_TYPE]: 'QUERY_TEXT'
      },
    },
  },
};
const storeResourceFill = {
  form: {
    clinicalQuerySearchForm: {
      values: {
        [valuesNames.SELECT_AGE]: 'birthday',
        [valuesNames.AGE_RANGE]: [
          0,
          100,
        ],
        [valuesNames.MALE]: true,
        [valuesNames.FEMALE]: true,
        [valuesNames.DATE_OF_BIRTH]: '28/09/1937',
        [valuesNames.QUERY_CONTAINS]: true,
        [valuesNames.SEARCH_TYPE]: 'allergies',
        [valuesNames.SEARCH_TYPE]: 'QUERY_TEXT'
      },
    },
  },
};
const store = mockStore(storeResource);
const storeFill = mockStore(storeResourceFill);
const context = {
  router: {
    history: {
      test: 'test',
      goBack: () => {},
      push: () => {},
      replace: () => {},
    },
  },
};

const simulateEvent = {
  preventDefault: () => {},
};

describe('Component <ClinicalQuerySearch />', () => {
  it('should renders with init data in store correctly', () => {
    const component = shallow(
      <ClinicalQuerySearch
        store={store}
      />, { context }).dive();
    component.setContext(context);
    expect(component.state().isOpen).toEqual(true);
    component.find('.btn-success').at(1).simulate('click', simulateEvent);
    expect(component.state().isOpen).toEqual(false);
    expect(component).toMatchSnapshot();

    component.setProps({ formIsValid: false });
    component.find('.btn-success').at(1).simulate('click', simulateEvent);
    expect(component).toMatchSnapshot();
  });

  it('should renders with fill Data in store correctly', () => {
    const component = shallow(
      <ClinicalQuerySearch
        store={storeFill}
      />, { context }).dive();
    component.setContext(context);
    expect(component.state().isOpen).toEqual(true);
    component.find('.btn-success').at(1).simulate('click', simulateEvent);
    expect(component.state().isOpen).toEqual(false);
    component.find('.btn-success').at(0).simulate('click', simulateEvent);
    expect(component.state().isOpen).toEqual(true);
    expect(component).toMatchSnapshot();
  });

});

