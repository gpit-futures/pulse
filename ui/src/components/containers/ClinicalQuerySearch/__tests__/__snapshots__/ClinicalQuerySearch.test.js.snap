// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <ClinicalQuerySearch /> should renders with fill Data in store correctly 1`] = `
<div
  className="dropdown-menu dropdown-menu-search dropdown-menu-right without-shadow"
>
  <div
    className="panel-group accordion"
  >
    <div
      className="panel panel-secondary without-margin open"
    >
      <div
        className="panel-heading"
      >
        <div
          className="control-group right"
        >
          <button
            className="btn btn-success btn-inverse btn-square btn-toggle-rotate"
            onClick={[Function]}
          >
            <i
              className="btn-icon fa fa-chevron-up"
            />
          </button>
        </div>
        <h3
          className="panel-title"
        >
          <span
            className="ng-binding"
          >
            Clinical Query
          </span>
          <span
            className="hidden-xs hidden-sm ng-binding"
          >
            : Search Type: QUERY_TEXT, Date of Birth: 28/09/1937, Gender: All
          </span>
        </h3>
      </div>
      <div
        className="panel-body"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <div
          className="panel-body-inner"
        >
          <ReduxForm
            formValues={
              Object {
                "ageRange": Array [
                  0,
                  100,
                ],
                "dateOfBirth": "28/09/1937",
                "queryContains": true,
                "selectAgeField": "birthday",
                "sexFemale": true,
                "sexMale": true,
                "type": "QUERY_TEXT",
              }
            }
            handleSearch={[Function]}
          />
        </div>
        <div
          className="panel-control"
        >
          <div
            className="wrap-control-group hide-indent-bottom"
          >
            <div
              className="control-group with-indent right"
            >
              <button
                className="btn btn-danger btn-icon-normal"
              >
                <i
                  className="btn-icon fa fa-times"
                />
                 
                <span
                  className="btn-text"
                >
                  Close
                </span>
              </button>
              <button
                className="btn btn-success btn-icon-normal"
                disabled={false}
                onClick={[Function]}
              >
                <i
                  className="btn-icon fa fa-search"
                />
                <span
                  className="btn-text"
                >
                  Search
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <ClinicalQuerySearch /> should renders with init data in store correctly 1`] = `
<div
  className="dropdown-menu dropdown-menu-search dropdown-menu-right"
>
  <div
    className="panel-group accordion"
  >
    <div
      className="panel panel-secondary without-margin"
    >
      <div
        className="panel-heading"
      >
        <div
          className="control-group right"
        >
          <button
            className="btn btn-success btn-inverse btn-square btn-toggle-rotate"
            onClick={[Function]}
          >
            <i
              className="btn-icon fa fa-chevron-up"
            />
          </button>
        </div>
        <h3
          className="panel-title"
        >
          <span
            className="ng-binding"
          >
            Clinical Query
          </span>
          <span
            className="hidden-xs hidden-sm ng-binding"
          >
            : Search Type: QUERY_TEXT, Age Range: 0-100
          </span>
        </h3>
      </div>
      <div
        className="panel-body"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="panel-body-inner"
        >
          <ReduxForm
            formValues={
              Object {
                "ageRange": Array [
                  0,
                  100,
                ],
                "queryContains": true,
                "selectAgeField": "range",
                "sexFemale": false,
                "sexMale": false,
                "type": "QUERY_TEXT",
              }
            }
            handleSearch={[Function]}
          />
        </div>
        <div
          className="panel-control"
        >
          <div
            className="wrap-control-group hide-indent-bottom"
          >
            <div
              className="control-group with-indent right"
            >
              <button
                className="btn btn-danger btn-icon-normal"
              >
                <i
                  className="btn-icon fa fa-times"
                />
                 
                <span
                  className="btn-text"
                >
                  Close
                </span>
              </button>
              <button
                className="btn btn-success btn-icon-normal"
                disabled={false}
                onClick={[Function]}
              >
                <i
                  className="btn-icon fa fa-search"
                />
                <span
                  className="btn-text"
                >
                  Search
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component <ClinicalQuerySearch /> should renders with init data in store correctly 2`] = `
<div
  className="dropdown-menu dropdown-menu-search dropdown-menu-right"
>
  <div
    className="panel-group accordion"
  >
    <div
      className="panel panel-secondary without-margin"
    >
      <div
        className="panel-heading"
      >
        <div
          className="control-group right"
        >
          <button
            className="btn btn-success btn-inverse btn-square btn-toggle-rotate"
            onClick={[Function]}
          >
            <i
              className="btn-icon fa fa-chevron-up"
            />
          </button>
        </div>
        <h3
          className="panel-title"
        >
          <span
            className="ng-binding"
          >
            Clinical Query
          </span>
          <span
            className="hidden-xs hidden-sm ng-binding"
          >
            : Search Type: QUERY_TEXT, Age Range: 0-100
          </span>
        </h3>
      </div>
      <div
        className="panel-body"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div
          className="panel-body-inner"
        >
          <ReduxForm
            formValues={
              Object {
                "ageRange": Array [
                  0,
                  100,
                ],
                "queryContains": true,
                "selectAgeField": "range",
                "sexFemale": false,
                "sexMale": false,
                "type": "QUERY_TEXT",
              }
            }
            handleSearch={[Function]}
          />
        </div>
        <div
          className="panel-control"
        >
          <div
            className="wrap-control-group hide-indent-bottom"
          >
            <div
              className="control-group with-indent right"
            >
              <button
                className="btn btn-danger btn-icon-normal"
              >
                <i
                  className="btn-icon fa fa-times"
                />
                 
                <span
                  className="btn-text"
                >
                  Close
                </span>
              </button>
              <button
                className="btn btn-success btn-icon-normal"
                disabled={true}
                onClick={[Function]}
              >
                <i
                  className="btn-icon fa fa-search"
                />
                <span
                  className="btn-text"
                >
                  Search
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
